### ISSUE TITLE
Error: long vectors not supported yet: qap_encode.c:36

### ISSUE THREAD

**Issue by NorthFuture** [s-u/Rserve#79]
In certain circumstance, randomly we have this error in Rserve. The error is not fully reproducible, and the only solution is to restart the Rserve. 

The strange thing is that the error persists even if we open a new connection to Rserve.

The Rscript we are executing is pretty straightforward and simple, and the amount of data is not big (just few kb).

We are using Rmpfr and Hash libraries. 

To connect to Rserve from java we are using org.rosuda.REngine.Rserve package

We reproduced this error with the cran 1.7 version and with latest 1.8.4 version

**Comment by edineipiovesan** [comment]
Did you solve the problem? I have a very similar problem with this error message.

**Comment by Stongtong** [comment]
I also have this error, some idea?

**Comment by NorthFuture** [comment]
Hi,
sorry for the late reply, but I've been travelling.

Unfortunately we couldn't solve the problem: we tried several things, from
removing and completly cleaning the workspace between each invocation, to
opening new connections to Reserve.

No matter what, we encountered this error from time to time. We understood
if we encounter this message, the entire Rservice process is messed up and
the only solution is to restart it.

What we did, is far from being ideal...but was the only way to get this
work. Basically we monitor rserve.log file and as soon as we find that
error message we restart the Rserve.

In our architecture we have a bunch of Rserve instances that can be used by
applications, so even if a Rserve goes down, other instances will serve
incoming requests.

Moreover, our applications are designed for resilience, so even if a call
fails, the application will try again (few more times) and eventually we
get the results.

The key is retrying.

The code for Rserve monitoring is
public static void main(String[] args) throws Exception {
        Process tailProcess = Runtime.getRuntime().exec("tail -f
/var/log/rserve/rserve.log");

        try (BufferedReader reader = new BufferedReader(new
InputStreamReader(tailProcess.getInputStream()))) {
            String line = null;
            while ((line = reader.readLine()) != null) {
                if (line.contains("long vectors not supported yet:
qap_encode.c:36")) {

                    System.out.println("Restarting");
                    Process process = Runtime.getRuntime().exec("service
Rserve stop");

                    process.waitFor();

                    Thread.sleep(500);

                    process = Runtime.getRuntime().exec("service Rserve
start");

                    process.waitFor();
                }
            }
        } catch (IOException e) {
            System.out.println(e.getMessage());
            throw new RuntimeException(e);
        }
    }


Hope it helps.


Il 11 Dic 2017 12:59 PM, "Edinei Piovesan" <notifications@github.com> ha
scritto:

**Comment by Stongtong** [comment]
ok, thank you. now I am  processing it by catch exception . Thank you for recive again.



发自我的小米手机
在 NorthFuture <notifications@github.com>，2017年12月13日 16:10写道：

Hi,
sorry for the late reply, but I've been travelling.

Unfortunately we couldn't solve the problem: we tried several things, from
removing and completly cleaning the workspace between each invocation, to
opening new connections to Reserve.

No matter what, we encountered this error from time to time. We understood
if we encounter this message, the entire Rservice process is messed up and
the only solution is to restart it.

What we did, is far from being ideal...but was the only way to get this
work. Basically we monitor rserve.log file and as soon as we find that
error message we restart the Rserve.

In our architecture we have a bunch of Rserve instances that can be used by
applications, so even if a Rserve goes down, other instances will serve
incoming requests.

Moreover, our applications are designed for resilience, so even if a call
fails, the application will try again (few more times) and eventually we
get the results.

The key is retrying.

The code for Rserve monitoring is
public static void main(String[] args) throws Exception {
Process tailProcess = Runtime.getRuntime().exec("tail -f
/var/log/rserve/rserve.log");

try (BufferedReader reader = new BufferedReader(new
InputStreamReader(tailProcess.getInputStream()))) {
String line = null;
while ((line = reader.readLine()) != null) {
if (line.contains("long vectors not supported yet:
qap_encode.c:36")) {

System.out.println("Restarting");
Process process = Runtime.getRuntime().exec("service
Rserve stop");

process.waitFor();

Thread.sleep(500);

process = Runtime.getRuntime().exec("service Rserve
start");

process.waitFor();
}
}
} catch (IOException e) {
System.out.println(e.getMessage());
throw new RuntimeException(e);
}
}


Hope it helps.


Il 11 Dic 2017 12:59 PM, "Edinei Piovesan" <notifications@github.com> ha
scritto:
